apps/api/app/services/workflow/context_extractor.py

```
@@ -0,0 +1,273 @@
+        # Also handle CUSTOM_ prefix
+        if name.upper().startswith("CUSTOM_"):
+            name = name[7:]
+
+        return name.replace("_", " ").title()
+
+    @classmethod
+    def _build_description(
+        cls,
+        tool_name: str,
+        args: Dict[str, Any],
+        output: str,
```

Ruff (0.14.14)

[warning] Unused class method argument: `output`

(ARG003)